tests/v2/validation/public_api/crdgeneration/crd_generation_test.go

```diff
@@ -0,0 +1,96 @@
+package crdgeneration
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CRDGenTestSuite struct {
+	suite.Suite
+	client      *rancher.Client
+	session     *session.Session
+	cluster     *management.Cluster
+	clusterName string
+	clusterID   string
+	crds        map[string][]string
+	clusterAuth bool
+	clusterV1   *v1.Cluster
+}
+
+func (crd *CRDGenTestSuite) TearDownSuite() {
+	crd.session.Cleanup()
+}
+
+func (crd *CRDGenTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	crd.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(crd.T(), err)
+
+	crd.client = client
+	crd.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(crd.T(), crd.clusterName, "Cluster name to install should be set")
+	crd.clusterID, err = clusters.GetClusterIDByName(crd.client, crd.clusterName)
+	require.NoError(crd.T(), err, "Error getting cluster ID")
+	crd.cluster, err = crd.client.Management.Cluster.ByID(crd.clusterID)
+	require.NoError(crd.T(), err)
+
+	readJson, err := os.ReadFile("../resources/crds.json")
+	require.NoError(crd.T(), err)
+	err = json.Unmarshal(readJson, &crd.crds)
+	require.NoError(crd.T(), err)
+
+	clusterType, err := clusters.NewClusterMeta(crd.client, crd.clusterName)
+	require.NoError(crd.T(), err)
+	//Change "local" to isLocal for Q4
+	if crd.clusterName != "local" && clusterType.Provider == clusters.KubernetesProviderRKE {
```

Comment:
I think this would be easier to read as:
```
crd.clusterAuth = false
if crd.clusterName != "local" && clusterType.Provider == clusters.KubernetesProviderRKE && crd.cluster.LocalClusterAuthEndpoint != nil && crd.cluster.LocalClusterAuthEndpoint.Enabled:
  crd.clusterAuth = true
```

Reply:
I initially went with this style but later changed it to the current style for its brevity. IMO adding this in a single line would be difficult to read.  @igomez06 @caliskanugur any suggestions?

Reply:
I don't have a strong opinion, I would be ok with you implementation Anu